test/e2e/utils.go

```diff
@@ -0,0 +1,252 @@
+}
+
+func waitForContourStatusConditions(t *testing.T, ctx context.Context, cl client.Client, timeout time.Duration, name, ns string, conditions ...metav1.Condition) error {
+	nsName := types.NamespacedName{Namespace: ns, Name: name}
+	return wait.PollImmediate(1*time.Second, timeout, func() (bool, error) {
+		cntr := &operatorv1alpha1.Contour{}
+		if err := cl.Get(ctx, nsName, cntr); err != nil {
+			t.Logf("failed to get contour %s/%s: %v", nsName.Namespace, nsName.Name, err)
+			return false, nil
+		}
+		expected := contourConditionMap(conditions...)
+		current := contourConditionMap(cntr.Status.Conditions...)
+		return contourConditionsMatchExpected(expected, current), nil
+	})
+}
+
+func waitForDeploymentStatusConditions(t *testing.T, ctx context.Context, cl client.Client, timeout time.Duration, name, ns string, conditions ...appsv1.DeploymentCondition) error {
+	nsName := types.NamespacedName{
+		Namespace: ns,
+		Name:      name,
+	}
+	return wait.PollImmediate(1*time.Second, timeout, func() (bool, error) {
+		deploy := &appsv1.Deployment{}
+		if err := cl.Get(ctx, nsName, deploy); err != nil {
+			t.Logf("failed to get deployment %s/%s: %v", nsName.Namespace, nsName.Name, err)
+			return false, nil
+		}
+		expected := deploymentConditionMap(conditions...)
+		current := deploymentConditionMap(deploy.Status.Conditions...)
+		return deploymentConditionsMatchExpected(expected, current), nil
+	})
+}
+
+func contourConditionMap(conditions ...metav1.Condition) map[string]string {
+	conds := map[string]string{}
+	for _, cond := range conditions {
+		conds[cond.Type] = string(cond.Status)
+	}
+	return conds
+}
+
+func deploymentConditionMap(conditions ...appsv1.DeploymentCondition) map[appsv1.DeploymentConditionType]corev1.ConditionStatus {
+	conds := map[appsv1.DeploymentConditionType]corev1.ConditionStatus{}
+	for _, cond := range conditions {
+		conds[cond.Type] = cond.Status
+	}
+	return conds
+}
+
+func contourConditionsMatchExpected(expected, actual map[string]string) bool {
+	filtered := map[string]string{}
+	for k := range actual {
+		if _, comparable := expected[k]; comparable {
+			filtered[k] = actual[k]
+		}
+	}
+	return reflect.DeepEqual(expected, filtered)
+}
+
+func deploymentConditionsMatchExpected(expected, actual map[appsv1.DeploymentConditionType]corev1.ConditionStatus) bool {
+	filtered := map[appsv1.DeploymentConditionType]corev1.ConditionStatus{}
+	for k := range actual {
+		if _, comparable := expected[k]; comparable {
+			filtered[k] = actual[k]
+		}
+	}
+	return reflect.DeepEqual(expected, filtered)
+}
+
+func waitForCurl(t *testing.T, url string, timeout time.Duration) {
+	err := wait.PollImmediate(1*time.Second, timeout, func() (bool, error) {
+		req, _ := http.NewRequest("GET", url, nil)
+		_, err := http.DefaultClient.Do(req)
+		if err != nil {
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		t.Fatalf("failed to curl %s: %v", url, err)
+	}
+	t.Logf("curl'd url %s", url)
```

Comment:
These errors are misleading since `curl` isn't actually involved. Suggest something similar to:

```Go
t.Logf("%s %q returned status %d", method, url, status)
```

same for the `Fatalf` above